src/components/Daybutton.js

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from "react";
+
+import styled from "styled-components";
+
+export default function Daybutton({ day, loading, habitdays, id }){
+    const [color, setColor] = useState("#DBDBDB");
+    const [backcolor, setBackcolor] = useState("#FFFFFF");
+    const [changecolor, setChangecolor] = useState(false);
+
+    useEffect(() => {
+        const userhabit = localStorage.getItem('habits');
+
+        if(userhabit !== null){
+            const data = JSON.parse(userhabit);
+            data.days.map((i) => {
+                if(i === id){
+                    return setChangecolor(true);
+                }
+            })
+        }
+    },[])
+
+    if(changecolor){
+        setColor("#FFFFFF");
+        setBackcolor("#CFCFCF");
+        setChangecolor(false);
+    }
+
+    function toggle(e){
+        e.preventDefault();
+        if (backcolor === "#FFFFFF"){
+            setColor("#FFFFFF");
+            setBackcolor("#CFCFCF");
+            habitdays("#FFFFFF", id);
+        }else if(backcolor === "#CFCFCF"){
+            setColor("#DBDBDB");
+            setBackcolor("#FFFFFF");
+            habitdays("#DBDBDB", id);
+        }
```

Comment:
**question**: Is there a possibility that the backcolor is neither of these two colors?

Reply:
Nop, :(